services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/credentials/AbstractCredentialsManagementService.java

```diff
@@ -44,6 +65,11 @@ public void setTenantInformationService(final TenantInformationService tenantInf
         this.tenantInformationService = tenantInformationService;
     }
 
+    @Autowired
+    public void setPasswordEncoder(final HonoPasswordEncoder passwordEncoder) {
+        this.passwordEncoder = passwordEncoder;
```

Comment:
does it make sense to allow setting this to `null`?

Reply:
I added `requireNonNull` check for tenant information service as well. We can introduce Noop version for folks that don't want to use autowired one.